doc: Document adjusting auto-scaling range with use_effective_fields (#4545)
mongodb/terraform-provider-mongodbatlas: docs/resources/advanced_cluster.md

This workflow allows you to set specific baseline values from which auto-scaling will resume dynamic adjustments based on workload.

### Adjusting the Auto-Scaling Range with use_effective_fields

> Changing the compute auto-scaling range (`compute_min_instance_size` or `compute_max_instance_size`) requires extra care when the new range would no longer contain the cluster's **current effective** instance size, for example raising `compute_min_instance_size` above the effective size or lowering `compute_max_instance_size` below it. Atlas validates the new range against the current effective instance size, so attempting either in a single apply fails with a validation error from Atlas. Updating `instance_size` in the same apply does not avoid the error, because the configured `instance_size` is not applied while auto-scaling is enabled with `use_effective_fields = true`.

Reviewer: ```suggestion
Changing the compute auto-scaling range (`compute_min_instance_size` or `compute_max_instance_size`) requires extra care when the new range would no longer contain the cluster's **current effective** instance size. This includes:

- Raising `compute_min_instance_size` above the effective size.
- Lowering `compute_max_instance_size` below the effective size.

Atlas validates the new range against the current effective instance size, so attempting either in a single apply fails with a validation error from Atlas. Updating `instance_size` in the same apply as changing the compute auto-scaling range does not avoid the error, because the updated `instance_size` value is not applied while auto-scaling is enabled with `use_effective_fields = true`.
```

Author: applied

> To adjust the range in this case, first move the effective instance size into the new range, then change the bounds:

Reviewer: ```suggestion
To adjust the range when the new range would no longer contain the cluster's **current effective** instance size, move the effective instance size into the new range before changing the bounds:
```

Author: Adjusted, used "would exclude the current effective instance size" to avoid repeating the full phrase from the bullets just above.

> 1. Temporarily disable compute auto-scaling (`compute_enabled = false`) and set `instance_size` to a value inside your intended new range, then apply. See [Manually Updating Specs with use_effective_fields](#manually-updating-specs-with-use_effective_fields) for details.

Reviewer 2: q: don't we need to disable all auto-scaling including disk? as an example changes in disk_iops or disk_size_gb might produce changes in instance_size.    see "manual updating..." section above:
```
1. Set `compute_enabled = false` and `disk_gb_enabled = false`  ...
```

might also want to just refer to that section instead of restating the disable/set-instance_size instructions here

Author: 👍  Step 1 now defers to the `Manually Updating Specs` workflow, which disables both compute and disk auto-scaling and covers the oplog_min_retention_hours requirement. Removed the restated disable/set instructions here.

> 1. Temporarily disable compute auto-scaling (`compute_enabled = false`) and set `instance_size` to a value inside your intended new range, then apply. See [Manually Updating Specs with use_effective_fields](#manually-updating-specs-with-use_effective_fields) for details.

Reviewer: ```suggestion
1. Disable compute auto-scaling (`compute_enabled = false`) and set `instance_size` to a value inside your intended new range, then apply. See [Manually Updating Specs with use_effective_fields](#manually-updating-specs-with-use_effective_fields) for details.
```

Author: applied

> 2. Re-enable compute auto-scaling (`compute_enabled = true`) with the new `compute_min_instance_size` and `compute_max_instance_size` bounds, then apply.

Reviewer: ```suggestion
2. Enable compute auto-scaling (`compute_enabled = true`) with the new `compute_min_instance_size` and `compute_max_instance_size` bounds, then apply.
```

Author: Prefer "Re-enable" since auto-scaling was on before step 1 disabled it. I did drop (compute_enabled = true) and made it "re-enable auto-scaling" generically, so disk auto-scaling is restored too if it was disabled in step 1.

> The same process applies to the [`analytics_auto_scaling`](#analytics_auto_scaling) block: disable `analytics_auto_scaling.compute_enabled` and set `analytics_specs.instance_size` in step 1.

Reviewer: Since the procedure is pretty short, it could help to write out the steps.

```suggestion
The same process applies to the [`analytics_auto_scaling`](#analytics_auto_scaling) block: 

1. Disable `analytics_auto_scaling.compute_enabled` and set `analytics_specs.instance_size` to a value inside your intended new range.
```

Author: Kept it as a one-line reference rather than repeating the numbered steps, since the procedure is identical to the compute one.

> -> **NOTE:** This multi-step process is only required while the current effective instance size would fall outside the new range. If the new range still contains the effective size, you can change the bounds in a single apply. This requirement may be removed in a future Atlas release, after which adjusting the range in a single apply will be supported.

Reviewer: ```suggestion
-> **NOTE:** This multi-step process is only required while the current effective instance size would fall outside the new range. If the new range still contains the effective size, you can change the bounds in a single apply.
```

We try to not publicize potential future updates if that's okay!

Author: 👍 removed
